src/services/authService.ts

```diff
@@ -23,29 +22,33 @@ export class AuthService implements IAuthService {
     async signInWithGoogle(): Promise<User | undefined> {
         try {
             const result = await signInWithPopup(this.auth, this.provider)
-            const userResponse = parseUser(result.user)
-            return await this.createUserOnFirstLogin(userResponse)
+            return this.retrieveUser(result.user)
```

Comment:
I've realized this sign-in function got a little confusing with the create user on the first login part. I've tried to refactor it to make it easier to understand.
The first step was to return a `retrieveUser` function instead of a `createUserOnFirstLogin` since we've not actually checked yet if it is the first login or not.

Reply:
I see, but this way I think retrieveUser does more than one think. At first It looks like by its name that it'll just fetch for this user in the database, but by looking at the function, it does 2 things, it checks if the user exists, then it creates it it not. 
I think you could break it in two pieces: 
 - function the check if the user is in the db (name suggestion: checkIfUserExists)
 - function to create the user (essentially, it does the same as the createUserOnFirstLogin, but it could be renamed to just createUser as this function could have more usage than just on first login ). 
 
 Those functions can lay inside the signInWithGoogle. I feel like it will expose better to the reader what signInWithGoogle does step by step this way